include/toolchain/gcc.h

```diff
@@ -59,6 +59,16 @@
 	__attribute__((section(#segment ",\"wa\",@nobits#")))
 #endif
 
+/**
+ * Round the provided value up to the nearest power of two
+ *
+ * @param x Value to round up
+ * @return Provided value rounded up to next power of two
+ */
+#define POW2_CEIL(x) ((1 << (31 - __builtin_clz(x))) < x ?  \
+		1 << (31 - __builtin_clz(x) + 1) : \
+		1 << (31 - __builtin_clz(x)))
```

Comment:
(http://locklessinc.com/articles/next_pow2/ might be interesting too. :)

Reply:
I would rather leave this exactly as-is, I just moved the definition.
Currently, if X is a constant, then this resolves to a constant at build time.